### PR TITLE
Week5part2

### PR DESCRIPTION
**Pull request by rivki770**

### CODE REVIEW

File: cup_of_join.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+def join(*list_param, sep="-"):
```
Comment (itayday) [inline cup_of_join.py:1]: 1. Please add documantation.
2. Is there a way to simplify the function? (for example, by using list comprehension?)

File: cut_a_cake.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+def get_recipe_price(prices, optionals=[], **ingredients):
```
Comment (itayday) [inline cut_a_cake.py:1]: Please add documantation to the function

File: cut_a_cake.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    cost = 0
+    for i in ingredients:
```
Comment (itayday) [inline cut_a_cake.py:3]: Try using a different name instead of "i" that will be more intuitive.
Same for "cost".

File: cup_of_join.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    if len(list_param) == 0:
+        return None
+
+    list_new = []
```
Comment (talbarfm) [inline cup_of_join.py:5]: list_new is a bad variable name

File: cup_of_join.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
+def join(*list_param, sep="-"):
+    if len(list_param) == 0:
+        return None
```
Comment (talbarfm) [inline cup_of_join.py:3]: Why return None and not an empty list, or throw an exception ?

File: cup_of_join.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+def main():
+    print(join([1, 2], [8], [9, 5, 6], sep='@'))
+    print(join([1, 2], [8], [9, 5, 6]))
+    print(join([1]))
+    print(join())
+
+
+if __name__ == '__main__' :
+    main()
```
Comment (talbarfm) [inline cup_of_join.py:21]: Add a newline at eof

File: cup_of_join.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
+def join(*list_param, sep="-"):
+    if len(list_param) == 0:
+        return None
+
+    list_new = []
+    for param in list_param:
```
Comment (talbarfm) [inline cup_of_join.py:6]: Try to use list comprehension with multiple iterators instead of this for loop

File: cut_a_cake.py [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+
+
+def main():
+    print(get_recipe_price({'chocolate': 18, 'milk': 8}, chocolate=200, milk=100))
+    print(get_recipe_price({'chocolate': 18, 'milk': 8}, optionals=['milk'], chocolate=300))
+    print(get_recipe_price({}))
+
+
+if __name__ == '__main__' :
+    main()
```
Comment (talbarfm) [inline cut_a_cake.py:16]: Add a newlist in end of file

File: cut_a_cake.py [hunk 9]
```diff
@@ -0,0 +1,16 @@
+def get_recipe_price(prices, optionals=[], **ingredients):
+    cost = 0
+    for i in ingredients:
+        if i not in optionals:
+            cost += prices[i] * (ingredients[i] / 100)
+    return int(cost)
+
+
+def main():
+    print(get_recipe_price({'chocolate': 18, 'milk': 8}, chocolate=200, milk=100))
```
Comment (talbarfm) [inline cut_a_cake.py:10]: Use " for strings unless the string contains "

File: cut_a_cake.py (hunk 3 above)
Comment (talbarfm) [inline cut_a_cake.py:3]: You can use reduce to do this

**Review by talbarfm — CHANGES_REQUESTED** [review]

**Pull request closed without merging**